v2/product/product-type.rst

```diff
@@ -1,4 +1,68 @@
 Create a product type
```

Comment:
We need an overview for this section. Give us 2-3 sentence summary of what we're accomplishing and why someone would want to create product types.